### ISSUE TITLE
Dataset Harvard 30K processing

### ISSUE THREAD

**Issue by Leesoon1984** [Qinkaiyu/RIMA#2]
How do you process the dataset Harvard 30K? Can you provide the scrips?

**Comment by Qinkaiyu** [comment]
Thanks for your interest in the data processing for the Harvard 30K dataset. the image processing is from the repository https://github.com/Chenngzz/IMDR. I'm not sure if I can share the processed images with you, so please follow this link to process them. We have deleted the low-quality images, and I can share the idx of the images we used with you if you need it.

**Comment by Leesoon1984** [comment]
> Thanks for your interest in the data processing for the Harvard 30K dataset. the image processing is from the repository https://github.com/Chenngzz/IMDR. I'm not sure if I can share the processed images with you, so please follow this link to process them. We have deleted the low-quality images, and I can share the idx of the images we used with you if you need it.

Great job, thank you for your sharing. We will replicate this work and make improvements based on it, so we need you to share the image IDs used. Thank you again.

**Comment by Qinkaiyu** [comment]
Sorry for the long wait, I went on a trip last week. This link https://drive.google.com/drive/folders/1zozblD5lVTcstsx3E6P7zOAxTr4w1iZV?usp=share_link
is the idx of the AMD and Glocoma datasets, and the dr is on my computer in the UK. I will upload the dr when I return. The training set and test set are divided based on random numbers. The training set and test volume may be different on Linux and Windows systems, so it is recommended that you retrain the model for fair comparison. Thank you for your attention. If you have any questions, please feel free to contact me!

**Comment by yinpub** [comment]
The author has deleted the repository of that paper (IMDR).  https://github.com/Chenngzz/IMDR Could you provide it in this repo?

**Comment by Qinkaiyu** [comment]
https://drive.google.com/drive/folders/1ksvWGp1nx3sykBrW7AHZVzFckocmYm2-?usp=drive_link Here is the link of datasets. If you have any further questions, plz feel free to contact me.

**Comment by yinpub** [comment]
Thank you. We just need data pre-process part. Could you tell me where is it? I hear that you have uploaded it.

**Comment by Qinkaiyu** [comment]
Okay, I've uploaded the preprocessed code. data_process.py

**Comment by seunghub** [comment]
Hi, thanks for sharing this repo.

I have a question regarding how to set the dataset-related arguments in data_process.py, specifically when using the official dataset.

In data_process.py, there are variables like:

source_folder = "/path/Test"
fundus_folder = "/path/fundus"
labels_fundus_file = "/path/fundus.txt

Could you clarify what each of these should correspond to in the official dataset structure?

Additionally, is there any way to directly download a preprocessed version of the Harvard dataset, or any officially supported preprocessing script/output that matches this code?

Any guidance or examples would be greatly appreciated.
Thank you!

**Comment by Qinkaiyu** [comment]
> Hi, thanks for sharing this repo.
> 
> I have a question regarding how to set the dataset-related arguments in data_process.py, specifically when using the official dataset.
> 
> In data_process.py, there are variables like:
> 
> source_folder = "/path/Test" fundus_folder = "/path/fundus" labels_fundus_file = "/path/fundus.txt
> 
> Could you clarify what each of these should correspond to in the official dataset structure?
> 
> Additionally, is there any way to directly download a preprocessed version of the Harvard dataset, or any officially supported preprocessing script/output that matches this code?
> 
> Any guidance or examples would be greatly appreciated. Thank you!
Hi, thanks for your message and for taking a close look at the repository.

Regarding the arguments in data_process.py:

source_folder generally refers to the root directory of the original dataset.

fundus_folder corresponds to the fundus image modality within the dataset.

Similarly, the OCT modality is handled separately (please see the section “# Harvard-30k Data Process3” in the code for the OCT-related paths and logic).

labels_fundus_file should point to the label file associated with the fundus images, following the format expected by the script.

That said, I should clarify that I personally worked with a preprocessed version of the Harvard-30k dataset. I did not have access to the fully raw official release at the time. The data processing script included in the repository was obtained from the original data processors, rather than being written entirely from scratch by me. As a result, I may not be able to give a fully precise, end-to-end mapping for every field when starting strictly from the official raw dataset.

We recommend downloading the Harvard-30k dataset from the official source and performing the preprocessing accordingly. 
**https://drive.google.com/drive/folders/1ksvWGp1nx3sykBrW7AHZVzFckocmYm2-?usp=sharing here is the link for the preprocessed version of the Harvard-30k dataset, that i used in my paper.**
You are also very welcome to evaluate and compare our method on other datasets of your interest.

If you encounter any issues, please feel free to contact me.
